### PR TITLE
Update b-tagger, add year configuring, and edit ntuple info

### PR DESCRIPTION
**Pull request by gmadigan**
Updated b-tagger to use Deep Flavor B (DeepJet). Added 'year' option to results producer for year-specific quantities, e.g., input trees, lumis, trigger paths. Adjusted 2016 and 2017 ntuple info (added some files, fixed some xsecs, event counts).

### CODE REVIEW

File: LQResultProducer.py [hunk 1]
```diff
@@ -9,31 +10,41 @@
...
+elif year == '2017':
+	NormalDirectory = '/eos/cms/store/group/phys_exotica/leptonsPlusJets/LQ/LQ2/stockNanoTrees/2017/gmadigan/NTupleAnalyzer_nanoAOD_Full2017QuickTest_stockNano_2020_05_04_21_47_24/SummaryFiles'
+	QCDDirectory    = '/eos/cms/store/group/phys_exotica/leptonsPlusJets/LQ/LQ2/trees/NTupleAnalyzer_nanoAOD_Full2016QCDNonIsoQuickTest_2019_10_14/SummaryFiles' #Placeholder
+	EMuDirectory    = '/eos/cms/store/group/phys_exotica/leptonsPlusJets/LQ/LQ2/trees/NTupleAnalyzer_nanoAOD_Full2016EMuSwitch_2019_10_14/SummaryFiles' #Placeholder
+elif year == '2018':
+	NormalDirectory = '/eos/cms/store/group/phys_exotica/leptonsPlusJets/LQ/LQ2/stockNanoTrees/2018/gmadigan/NTupleAnalyzer_nanoAOD_Full2018QuickTest_stockNano_2020_07_05_00_00_41/SummaryFiles'
+	QCDDirectory    = '/eos/cms/store/group/phys_exotica/leptonsPlusJets/LQ/LQ2/trees/NTupleAnalyzer_nanoAOD_Full2016QCDNonIsoQuickTest_2019_10_14/SummaryFiles' #Placeholder
+	EMuDirectory    = '/eos/cms/store/group/phys_exotica/leptonsPlusJets/LQ/LQ2/trees/NTupleAnalyzer_nanoAOD_Full2016EMuSwitch_2019_10_14/SummaryFiles' #Placeholder
+else: exit()
```
Comment (DavidMorse) [inline LQResultProducer.py]: Please add an error message before the exit that explains what options you failed to pick

File: LQResultProducer.py [hunk 2]
```diff
@@ -549,7 +555,6 @@ def main():
 		#munu2 = '(MT_uv>70)*(MT_uv<110)*(JetCount>3.5)'
 		[[Rz_uuj,Rz_uuj_err],[Rtt_uuj,Rtt_uuj_err]] = GetMuMuScaleFactors( NormalWeightMuMu+'*'+preselectionmumu_single, NormalDirectory, '(M_uu>80)*(M_uu<100)', '(M_uu>100)*(M_uu<250)',0,0)
 		[[Rz_uujj,Rz_uujj_err],[Rtt_uujj,Rtt_uujj_err]] = GetMuMuScaleFactors( NormalWeightMuMu+'*'+preselectionmumu, NormalDirectory, '(M_uu>80)*(M_uu<100)', '(M_uu>100)*(M_uu<250)',0,0)
-                #exit()
```
Comment (DavidMorse) [inline LQResultProducer.py:552]: This is a useful line, if you just want to check SF without making plots. I would leave it in, just commented out as it is here.

File: LQResultProducer.py [hunk 3]
```diff
@@ -5642,9 +5649,9 @@ def MakeBasicPlot(recovariable,xlabel,presentationbinning,selection,weight,FileD
 
 
 	MCStack=BeautifyStack(MCStack,Label)
-	hs_rec_Signal.Draw("HISTSAME")
-	if 'final' not in tagname:
-		hs_rec_Signal2.Draw("HISTSAME")
+	#hs_rec_Signal.Draw("HISTSAME")
```
Comment (DavidMorse) [inline LQResultProducer.py]: I don't think you want the signal commented out anymore, do you? We should have signal samples now, can start getting them on the plots.

File: LQResultProducer.py [hunk 4]
```diff
@@ -5642,9 +5649,9 @@ def MakeBasicPlot(recovariable,xlabel,presentationbinning,selection,weight,FileD
...
+	#if 'final' not in tagname:
```
Comment (DavidMorse) [inline LQResultProducer.py]: same here for signal

File: LQResultProducer.py [hunk 5]
```diff
@@ -5698,8 +5705,8 @@ def MakeBasicPlot(recovariable,xlabel,presentationbinning,selection,weight,FileD
 	if 'final' not in tagname:
 		if 'PAS' in tagname:
 			leg.AddEntry(hs_bgband,'Unc. (stat + syst)')
-		leg.AddEntry(hs_rec_Signal,sig1name,"l")
-		leg.AddEntry(hs_rec_Signal2,sig2name,"l")
+		#leg.AddEntry(hs_rec_Signal,sig1name,"l")
```
Comment (DavidMorse) [inline LQResultProducer.py]: same here for signal

**Review by DavidMorse — CHANGES_REQUESTED** [review]

File: LQResultProducer.py (hunk 3 above)
Comment (gmadigan) [reply LQResultProducer.py]: True! Added it in.

**Pull request merged**